test/example_data.dart

```diff
@@ -167,26 +178,44 @@ DmMessage dmMessage(
 
 // TODO example data for many more types
 
-final InitialSnapshot initialSnapshot = InitialSnapshot(
-  queueId: '1:2345',
-  lastEventId: 1,
-  zulipFeatureLevel: recentZulipFeatureLevel,
-  zulipVersion: recentZulipVersion,
-  zulipMergeBase: recentZulipVersion,
-  alertWords: ['klaxon'],
-  customProfileFields: [],
-  subscriptions: [], // TODO add subscriptions to example initial snapshot
-  streams: [], // TODO add streams to example initial snapshot
-  maxFileUploadSizeMib: 25,
-  realmUsers: [],
-  realmNonActiveUsers: [],
-  crossRealmBots: [],
-);
+InitialSnapshot initialSnapshot({
+  String? queueId,
+  int? lastEventId,
+  int? zulipFeatureLevel,
+  String? zulipVersion,
+  String? zulipMergeBase,
+  List<String>? alertWords,
+  List<CustomProfileField>? customProfileFields,
+  List<RecentDmConversation>? recentPrivateConversations,
+  List<Subscription>? subscriptions,
+  List<ZulipStream>? streams,
+  int? maxFileUploadSizeMib,
+  List<User>? realmUsers,
+  List<User>? realmNonActiveUsers,
+  List<User>? crossRealmBots,
+}) {
+  return InitialSnapshot(
+    queueId: queueId ?? '1:2345',
+    lastEventId: lastEventId ?? 1,
+    zulipFeatureLevel: zulipFeatureLevel ?? recentZulipFeatureLevel,
+    zulipVersion: zulipVersion ?? recentZulipVersion,
+    zulipMergeBase: zulipMergeBase ?? recentZulipVersion,
+    alertWords: alertWords ?? ['klaxon'],
+    customProfileFields: customProfileFields ?? [],
+    recentPrivateConversations: recentPrivateConversations ?? [], // TODO add recentPrivateConversations to default
```

Comment:
Makes sense.

Also, I realized after making this commit—

```
test [nfc]: Let callers customize initial snapshot from example data

This will be helpful when we add tests for the recent-DM-
conversations view model, coming up.
```

—that I didn't actually end up using `eg.initialSnapshot` for the new tests. `RecentDmConversationsView` doesn't take an `InitialSnapshot` as input; it takes a `List<RecentDmConversation>` that's supposed to be taken from the initial snapshot, and I just built a `List<RecentDmConversation>` directly.